### PR TITLE
[Serve] Implement metric interface

### PR DESCRIPTION
**Pull request by simon-mo**
Implement a metric interface that aggregate metrics from router and worker. 

```python
>>> serve.stat()
{
    "echo:v1_error_counter": 0.0,
    "echo:v1_latency_s_50th_perc_10_window": 0.0006570816040039062,
    "echo:v1_latency_s_50th_perc_300_window": 0.0006570816040039062,
    "echo:v1_latency_s_50th_perc_3600_window": 0.0006570816040039062,
   ...
}
```

### CODE REVIEW

File: python/ray/experimental/serve/global_state.py [hunk 1]
```diff
@@ -50,6 +52,9 @@ def __init__(self):
         #  use random/available port in a pre-defined port range. TODO(simon)
         self.http_address = ""
 
+        #: Metric monitor handle
```
Comment (pcmoritz) [inline python/ray/experimental/serve/global_state.py]: Why is there a `:` after the `#`?

File: python/ray/experimental/serve/metric.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+        self.gc_window_seconds = gc_window_seconds
+        self.latest_gc_time = time.time()
+
+    def add_target(self, target_handle):
+        self.actor_handles.append(target_handle)
+
+    def remove_target(self, target_handle):
+        found_idx = None
+        for i, handle in enumerate(self.actor_handles):
```
Comment (pcmoritz) [inline python/ray/experimental/serve/metric.py]: This looks like `self.actor_handles` should be a dict from `handle._ray_actor_id.hex()` to the handle :)

File: python/ray/experimental/serve/global_state.py (hunk 1 above)
Comment (simon-mo) [reply python/ray/experimental/serve/global_state.py]: This is sphinx's attribute notation
https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoattribute

Comment (pcmoritz) [reply python/ray/experimental/serve/global_state.py]: Ah ok I see!

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**